packages/trader/src/Modules/Trading/Components/Form/TradeParams/Turbos/barrier-selector.jsx

```diff
@@ -0,0 +1,153 @@
+            <p>
+                <span style={{ fontWeight: 'bold' }}>For Short: </span>You will earn profit if the market stays below
+                the entry spot and doesn&apos;t cross the barrier.
+            </p>
+        </React.Fragment>
+    );
+
+    const toggleBarriersTable = () => setIsBarriersTableExpanded(!is_barriers_table_expanded);
+
+    const changeBarrier = e => {
+        e.stopPropagation();
+        setHoveredBarrier(null);
+        setSelectedBarrier(e.target.id);
+        onChange({
+            target: {
+                name: 'barrier_1',
+                value: e.target.id,
+            },
+        });
+    };
+
+    const onMouseEnter = e => {
+        if (e.target.id && selected_barrier !== e.target.id) {
+            setHoveredBarrier(e.target.id);
+        }
+    };
+
+    const onMouseLeave = () => {
+        setHoveredBarrier(null);
+    };
+
+    React.useEffect(() => setSelectedBarrier(barrier_1), [barrier_1]);
+
+    return (
+        <React.Fragment>
+            {is_barriers_table_expanded && (
+                <CSSTransition
+                    appear
+                    in={is_barriers_table_expanded}
+                    timeout={250}
+                    classNames={{
+                        appear: 'trade-container__barriers-table--enter',
+                        enter: 'trade-container__barriers-table--enter',
+                        enterDone: 'trade-container__barriers-table--enter-done',
+                        exit: 'trade-container__barriers-table--exit',
+                    }}
+                    unmountOnExit
+                >
+                    <Fieldset className='trade-container__fieldset trade-container__barriers-table'>
+                        <div className='trade-container__barriers-table__header'>
+                            <Text color='prominent' weight='bold' size='xs'>
+                                {localize('Barriers')}
+                            </Text>
+                            <div className='trade-container__barriers-table__icon-close' onClick={toggleBarriersTable}>
+                                <Icon icon='IcCross' />
+                            </div>
+                        </div>
+                        <div className='trade-container__barriers-table__text'>Distance to spot</div>
+                        <ThemedScrollbars>
+                            <BarriersList
+                                base_classname='trade-container__barriers-table__item'
+                                active_item_classname='trade-container__barriers-table__item--selected'
+                                className='trade-container__barriers-table__list'
+                                list={turbos_barrier_choices}
+                                selected_item={selected_barrier}
+                                onClick={changeBarrier}
+                                onMouseLeave={onMouseLeave}
+                                onMouseEnter={onMouseEnter}
```

Comment:
@kate-deriv Let's incapsulate event handlers inside BarrierList and pass data to this parent using custom callbacks.  We can add a prop called e.g. onHover(..) to BarriersList which we will use as a callback to inform the parent about which barrier was hovered when onMouseLeave and onMouseEnter events fire from BarriersList 🙂 Same for onClick()